### PR TITLE
perf: a small stage validation optimization

### PR DESCRIPTION
**Pull request by Suor**

### CODE REVIEW

File: dvc/utils/__init__.py [hunk 1]
```diff
@@ -292,14 +292,22 @@ def fix_env(env=None):
...
-    if isinstance(data, builtin_str):
-        return str(data)
-    if isinstance(data, dict):
-        return dict(map(convert_to_unicode, data.items()))
-    if isinstance(data, (list, tuple)):
-        return type(data)(map(convert_to_unicode, data))
-    return data
+if builtin_str is str:
+
+    def convert_to_unicode(data):
```
Comment (efiop) [inline dvc/utils/__init__.py]: How does `data` depend on whether or not `builtin_str` is a str or not?

**Comment by efiop** [comment]
@Suor Thanks, @Suor ! Mind elaborating on this and showing how much this saves us?

File: dvc/utils/__init__.py (hunk 1 above)
Comment (Suor) [reply dvc/utils/__init__.py]: The function converts all `builtin_str` to `str` (imported from `utils.compat`) in a nested structure, which is a noop if they are the same, basically in Python 3.

**Comment by Suor** [comment]
@efiop I haven't benched it, maybe it's not worth it. And this way it's more obvious that we should drop it with Python 2. 

I came across this while optimizing schema validation, the [second slowest thing](https://github.com/iterative/dvc/issues/2671#issuecomment-549147265) in stage collection.

**Comment by efiop** [comment]
@Suor if this is python2 dependent, then put it into compat or something to make it more obvious.

**Comment by efiop** [comment]
@Suor Also, so this change is only speeding up python2, correct?

**Comment by Suor** [comment]
Moved it to compat.

**Pull request merged**